src/mkdocs_gallery/gen_single.py

```diff
@@ -767,7 +767,7 @@ def execute_code_block(compiler, block, script: GalleryScript):
 
     cwd = os.getcwd()
     # Redirect output to stdout
-    src_file = script.src_py_file.as_posix()
+    src_file = script.src_py_file
```

Comment:
`src_file` is used in many lines in this function body. In particular `_set_reset_logging_tee`, `os.chdir` and `compile`. I can not guarantee that these behave similarly when they receive a path object.

I am wondering if it would be safer to leave the `as_posix()` here and rename the variable `src_file_posix`. Then line 812 we would directly use `script.src_py_file` as the dict key.

An alternative otherwise would be to make sure that all places where `src_file` is used here explicitly allow a `Path` object to be received (in their signature type hint), and behave correctly with it.

Reply:
> An alternative otherwise would be to make sure that all places where `src_file` is used here explicitly allow a `Path` object to be received (in their signature type hint), and behave correctly with it.

If the objective is to eventually get rid of the `as_posix()` conversions, I'll vote for your 2nd suggestion.